control-plane/api/v1alpha1/api_gateway_types.go

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	GatewayClassConfigKind = "GatewayClassConfig"
+	managedLabel           = "api-gateway.consul.hashicorp.com/managed"
+	nameLabel              = "api-gateway.consul.hashicorp.com/name"
+	namespaceLabel         = "api-gateway.consul.hashicorp.com/namespace"
+	createdAtLabel         = "api-gateway.consul.hashicorp.com/created"
```

Comment:
Can we also drop these, IIRC there are already equivalent labels within the [codebase](https://github.com/hashicorp/consul-k8s/blob/0e771fcd49be3af400caec8da0f46e52ccd29b88/control-plane/connect-inject/constants/annotations_and_labels.go) that we can leverage. So I don't think these are needed -- the `GatewayClassConfigKind` we can keep because we'll need it really soon for doing checks of parameterRef types for `GatewayClass` objects.